src/plugins/vis_type_timeseries/server/usage_collector/get_usage_collector.ts

```diff
@@ -59,28 +74,8 @@ export const getStats = async (
     }
   }
 
-  if (esResponse?.hits?.hits?.length) {
-    for (const hit of esResponse.hits.hits) {
-      if (hit._source && 'visualization' in hit._source) {
-        const { visualization } = hit._source!;
-
-        let visState: VisState = {};
-        try {
-          visState = JSON.parse(visualization?.visState ?? '{}');
-        } catch (e) {
-          // invalid visState
-        }
-
-        telemetryUseLastValueMode(visState);
-      }
-    }
-  }
-
-  const byValueVisualizations = await findByValueEmbeddables(soClient, 'visualization');
-
-  for (const item of byValueVisualizations) {
-    telemetryUseLastValueMode(item.savedVis as VisState);
-  }
+  await doTelemetryFoVisualizations(soClient, telemetryUseLastValueMode);
+  await doTelemetryForByValueVisualizations(soClient, telemetryUseLastValueMode);
```

Comment:
I wonder if we could do run these queries concurrently. What do you think?

Reply:
why not? good point. PR was updated